### PR TITLE
Advanced settings update

### PR DESCRIPTION
**Pull request by nicodh**
This is an attempt to fix the update advanced settings implementation

### CODE REVIEW

File: scss/_global.scss [hunk 1]
```diff
@@ -180,3 +180,7 @@ code {
     background-color: unset;
   }
 }
+
+.login-form button[type='submit'] {
```
Comment (nicodh) [inline scss/_global.scss]: @Simon-Laux maybe move this to a better place - I am not so involved in css structure

File: src/main/deltachat/login.js [hunk 2]
```diff
@@ -48,13 +48,13 @@ module.exports = class DCLoginController extends SplitOut {
...
         this._controller.emit('ready', this._controller.credentials)
         log.info('dc_get_info', dc.getInfo())
-        render()
+        sendStateToRenderer()
       }
-      if (!this._dc.isConfigured()) {
+      if (!this._dc.isConfigured() || updateConfiguration) {
         this._dc.once('ready', onReady)
         this._controller.configuring = true
         this._dc.configure(addServerFlags(credentials))
```
Comment (nicodh) [inline src/main/deltachat/login.js]: Here I would expect, that the credentials are updated in the database but they are not...??

File: src/main/deltachat/settings.js [hunk 3]
```diff
@@ -27,4 +27,16 @@ module.exports = class DCSettings extends SplitOut {
   keysExport (directory) {
     this._dc.importExport(C.DC_IMEX_EXPORT_SELF_KEYS, directory)
   }
+
```
Comment (nicodh) [inline src/main/deltachat/settings.js]: with this method we could update the settings in the database, but it won't update the prefixed keys (configured_mail_user etc)

File: src/main/deltachat/login.js (hunk 2 above)
Comment (link2xt) [reply src/main/deltachat/login.js]: configure() function three lines below the change calls set_config.

File: src/main/deltachat/settings.js (hunk 3 above)
Comment (link2xt) [reply src/main/deltachat/settings.js]: Prefixed keys are updated by the core, there is no need to ever update them manually. If they are not updated, that means configuration didn't succeed

File: src/main/deltachat/login.js (hunk 2 above)
Comment (nicodh) [reply src/main/deltachat/login.js]: Yes but if I test it I don't get the updated values in the database...??

File: scss/_global.scss (hunk 1 above)
Comment (Simon-Laux) [reply scss/_global.scss]: currently login styles seem to be still in scss

Comment (Simon-Laux) [reply scss/_global.scss]: src/renderer/components/Login-Styles.js

**Comment by nicodh** [comment]
So what I found so far: if I uncomment this line https://github.com/deltachat/deltachat-node/blob/master/lib/deltachat.js#L99 advanced settings are updated except the server flags. For the server flags I see the value for "server_flags" updated as expected, but not for "configured_server_flags". No matter what the server_flags value is, the configured_server_flags value always is 66052 (DC_LP_IMAP_SOCKET_SSL & DC_LP_SMTP_SOCKET_STARTTLS).
Maybe a core bug?

File: src/main/ipc.js [hunk 4]
```diff
@@ -12,7 +12,7 @@ const localize = require('../localize')
 const menu = require('./menu')
 const windows = require('./windows')
 const log = require('../logger').getLogger('main/ipc', true)
-const DeltaChat = (() => {
+const DeltaChatController = (() => {
```
Comment (Simon-Laux) [inline src/main/ipc.js]: +1

File: scss/_global.scss (hunk 1 above)
Comment (Simon-Laux) [reply scss/_global.scss]: or you can already create a new scss file like scss/login/login-styles.scss
and don't forget to add the file to manifest.scss when you do that.
(you don't need to move everything there, only creating it for this class if fine for now)

**Comment by nicodh** [comment]
I also wonder why we need this temporary DC instance to test the login.
https://github.com/deltachat/deltachat-desktop/blob/master/src/main/ipc.js#L221
That's something DC core does anyway. If the new configured values are not working, the "configured_[keys]" in the DB  are not updated. So it's no problem to try to update the configuration immediately for the current account...

File: src/renderer/components/Login.js [hunk 5]
```diff
@@ -139,7 +147,7 @@ export default class Login extends React.Component {
     return (
       <React.Fragment>
         {this.renderLoginHeader(mode)}
-        <form onSubmit={this.handleSubmit}>
+        <form onSubmit={this.handleSubmit} className={'login-form'}>
```
Comment (Jikstra) [inline src/renderer/components/Login.js]: className='login-form' is enough

File: src/main/deltachat/settings.js [hunk 6]
```diff
@@ -2,6 +2,16 @@ const C = require('deltachat-node/constants')
 const log = require('../../logger').getLogger('main/deltachat/settings', true)
 
 const SplitOut = require('./splitout')
+
+const serverFlagMap = {
```
Comment (link2xt) [inline src/main/deltachat/settings.js]: I don't get what this map is for. Nobody uses strings like `mail_security_ssl` anywhere, they are never passed to `getConfigFor`.

What is the reason for this server flags refactoring?

Comment (nicodh) [reply src/main/deltachat/settings.js]: It's just for displaying the values in the Settings dialog. DC core uses the flags but humans need them to be translated into the according string values. So we get stings (like "ssl"or "plain") from the frontend and have to pass a bitmask to the dc core and vice versa

**Comment by link2xt** [comment]
Just tested, server flags configuration is fixed.

During login, "context not open" errors are displayed.

**Pull request merged**